src/array.cr

```diff
@@ -400,7 +400,9 @@ class Array(T)
   def []=(index : Int, count : Int, value : T)
     raise ArgumentError.new "Negative count: #{count}" if count < 0
 
-    index = check_index_out_of_bounds index
+    index += size if index < 0
+    raise IndexError.new unless 0 <= index <= size
```

Comment:
Could you add a comment to explain why not using check_index_out_of_bounds index?

Reply:
Good idea. Done!